builtin/providers/azure/config.go

```diff
@@ -0,0 +1,63 @@
+package azure
+
+import (
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/management"
+)
+
+// Config is a struct which holds all the required information to access
+// Azure services.
+type Config struct {
+	PublishSettingsFilePath string
+	SubscriptionID          string
+	ManagementCert          []byte
+	ManagementUrl           string
+}
+
+// AzureClient contains all the handles required for managing Azure services.
+type AzureClient struct {
+	// the client which holds the authentification information
+	managementClient management.Client
+
+	// unfortunately; because of how Azure's network API works; doing networking operations
+	// concurrently is very hazardous, and we need a mutex.
+	mutex *sync.Mutex
```

Comment:
Oh yes it does, trial & error being my witness.
The mutex is wholly necessary as per how the API works; you'd get into the awful jam in which two network resources would be in a sort-of 'creation race condition', in which the last one to send its network config back to Azure would be the only one which'd get created...